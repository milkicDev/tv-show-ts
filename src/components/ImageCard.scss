.ImageCard {
  font-weight: bold;
  max-width: 21rem;

  .rating {
    margin-top: 1rem;
    font-size: 2rem;
  }
}