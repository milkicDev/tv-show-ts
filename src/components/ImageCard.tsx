import React from 'react';
import { Link } from 'react-router-dom';
import Rating from 'react-rating';
import Show from '../models/Show';

import './ImageCard.scss';
import { getCalcRating } from '../helpers/functions';

type Props = {
  data: Show;
};

const ImageCard = (props: Props) => {
  return (
    <div className="ImageCard">
      <Link to={`/show/${props.data.id}`}>
        <img src={props.data.image?.medium} alt={props.data.name} />
      </Link>

      <Rating
        className='rating'
        readonly
        initialRating={getCalcRating(props.data?.rating.average)}
        emptySymbol='fa fa-star-o medium'
        fullSymbol='fa fa-star medium'
      />

      <p>{props.data.name}</p>
    </div>
  );
};

export default ImageCard;