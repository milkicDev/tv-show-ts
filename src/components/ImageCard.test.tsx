import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ShowsApi from '../api/Shows';
import Show from '../models/Show';
import ImageCard from './ImageCard';

it('render app', async () => {
  let render, show: Show;

  await ShowsApi.fetch(1)
    .then((res) => {
      show = res;
      render = renderer.create(
        <BrowserRouter>
          <ImageCard data={show} />
        </BrowserRouter>
        ).toJSON();
    });
    
  expect(render).toMatchSnapshot();
})
