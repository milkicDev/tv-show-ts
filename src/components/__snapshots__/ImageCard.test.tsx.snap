// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<div
  className="ImageCard"
>
  <a
    href="/show/1"
    onClick={[Function]}
  >
    <img
      alt="Under the Dome"
      src="https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg"
    />
  </a>
  <span
    className="rating"
    style={
      Object {
        "direction": "ltr",
        "display": "inline-block",
      }
    }
  >
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <span
          className="fa fa-star-o medium"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <span
          className="fa fa-star medium"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <span
          className="fa fa-star-o medium"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <span
          className="fa fa-star medium"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <span
          className="fa fa-star-o medium"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <span
          className="fa fa-star medium"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o medium"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "29.999999999999982%",
          }
        }
      >
        <span
          className="fa fa-star medium"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o medium"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star medium"
        />
      </span>
    </span>
  </span>
  <p>
    Under the Dome
  </p>
</div>
`;
